drivers/i2c/Kconfig

```diff
@@ -90,19 +90,36 @@ config I2C_NRF5_GPIO_SCL_PIN
 	help
 	  The GPIO pin to use for SCL.
 
-config I2C_STM32LX
-	bool "STM32Lx MCU I2C Driver"
-	depends on SOC_FAMILY_STM32 && SOC_SERIES_STM32L4X
+config I2C_STM32
```

Comment:
Is there are 2 different drivers, remove I2C_STM32, doesn't add any particular value.